### PR TITLE
NAS-111173 / 21.08 / add system.hostname endpoint and change callers to use it

### PR DESCRIPTION
**Pull request by yocalebo**

### CODE REVIEW

**Comment by yocalebo** [comment]
Note: rebased on https://github.com/truenas/middleware/pull/7073 so merge that one first before this one.

**Comment by yocalebo** [comment]
I tried to be clever with rebasing and don't really know what I'm doing. I don't like what this is doing so I'll close and re-open fresh.

File: src/middlewared/middlewared/plugins/system.py [hunk 1]
```diff
@@ -524,27 +524,39 @@ async def dmidecode_info(self):
             self.DMIDECODE_CACHE['ecc-memory'] = ecc
 
         if self.DMIDECODE_CACHE['baseboard-manufacturer'] is None:
-            bm = ((await run(["dmidecode", "-s", "baseboard-manufacturer"], check=False)).stdout.decode(errors='ignore')).strip()
+            bm = (
+                (await run(["dmidecode", "-s", "baseboard-manufacturer"], check=False)).stdout.decode(errors='ignore')
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/system.py:528]: Can we please be consistent with quotes ? Like use single or double but not both ?

File: src/middlewared/middlewared/plugins/system.py [hunk 2]
```diff
@@ -524,27 +524,39 @@ async def dmidecode_info(self):
...
+            ).strip()
             self.DMIDECODE_CACHE['baseboard-manufacturer'] = bm
 
         if self.DMIDECODE_CACHE['baseboard-product-name'] is None:
-            bpn = ((await run(["dmidecode", "-s", "baseboard-product-name"], check=False)).stdout.decode(errors='ignore')).strip()
+            bpn = (
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/system.py:533]: Is it useful to set all of these variables separately and have a separate `if` condition for each variable ? In other words, won't this only be true once ?

File: src/middlewared/middlewared/plugins/network.py [hunk 3]
```diff
@@ -1091,10 +1093,18 @@ def _get_filters(key):
             return [[key, '!=', update['id']]] if update else []
 
         validation_attrs = {
-            'aliases': ['Active node IP address', ' cannot be changed.', ' is required when configuring HA'],
-            'failover_aliases': ['Standby node IP address', ' cannot be changed.', ' is required when configuring HA'],
-            'failover_virtual_aliases': ['Virtual IP address', ' cannot be changed.', ' is required when configuring HA'],
-            'failover_group': ['Failover group number', ' cannot be changed.' ' is required when configuring HA'],
+            'aliases': [
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/network.py:1096]: Can you please clarify why we have changed this ?

File: src/middlewared/middlewared/plugins/network.py [hunk 4]
```diff
@@ -1325,10 +1335,9 @@ def _get_filters(key):
                 # newly created laggs.
                 if not update:
                     if data.get('failover_critical') and data.get('lag_protocol') == 'FAILOVER':
-                        verrors.add(
-                            f'{schema_name}.failover_critical',
-                            'A lagg interface using the "Failover" protocol is not allowed to be marked critical for failover.'
-                        )
+                        msg = 'A lagg interface using the "Failover" protocol '
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/network.py:1338]: Same here

File: src/middlewared/middlewared/plugins/network.py [hunk 5]
```diff
@@ -682,7 +682,9 @@ def iface_extend(self, iface_state, configs, ha_hardware, fake=False):
                     'netmask': int(config['int_v6netmaskbit']),
                 })
 
-        for alias in self.middleware.call_sync('datastore.query', 'network.alias', [('alias_interface', '=', config['id'])]):
+        for alias in self.middleware.call_sync(
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/network.py:685]: Same as below

File: src/middlewared/middlewared/plugins/network.py [hunk 6]
```diff
@@ -96,14 +96,16 @@ class Config:
             strict=True
         ),
         Str('hostname_local', required=True, validators=[Hostname()]),
+        Str('hostname_b', validators=[Hostname()], null=True),
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/network.py:99]: Can this be empty string ?

File: src/middlewared/middlewared/plugins/mail.py [hunk 7]
```diff
@@ -406,9 +405,13 @@ def read_json():
             msg['Cc'] = ', '.join(message.get('cc'))
         msg['Date'] = formatdate()
 
-        local_hostname = socket.gethostname()
-
-        msg['Message-ID'] = "<%s-%s.%s@%s>" % (sw_name.lower(), datetime.utcnow().strftime("%Y%m%d.%H%M%S.%f"), base64.urlsafe_b64encode(os.urandom(3)), local_hostname)
+        local_hostname = self.middleware.call_sync('system.hostname')
+        msg['Message-ID'] = "<%s-%s.%s@%s>" % (
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/mail.py:409]: Can we use f-string please ?

File: src/middlewared/middlewared/plugins/system.py (hunk 1 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/system.py:528]: That's how it was before but I agree, I will go back and fix it.

File: src/middlewared/middlewared/plugins/system.py (hunk 2 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/system.py:533]: Correct, this should only be set once for each item. This is static data that never changes after boot or after middlewared starts up for the first time.

File: src/middlewared/middlewared/plugins/network.py (hunk 3 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/network.py:1096]: flake8 fix one of the lines were too long but I changed all of them so they were all consistent.

File: src/middlewared/middlewared/plugins/network.py (hunk 4 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/network.py:1338]: flake8 fix for line being too long

File: src/middlewared/middlewared/plugins/system.py (hunk 2 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/system.py:533]: Yes but that would not happen here ;) This if would never be true twice

File: src/middlewared/middlewared/plugins/network.py (hunk 6 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/network.py:99]: This was the weird rebase issue I had, but this change has already been merged into master

File: src/middlewared/middlewared/plugins/network.py (hunk 3 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/network.py:1096]: We have merged this, but i think we should have flake8 changes in a different PR to limit the scope of a PR.

File: src/middlewared/middlewared/plugins/network.py (hunk 5 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/network.py:685]: flake8 fix for line being too long

**Pull request closed without merging**